lit_nlp/client/elements/text_diff.ts

```diff
@@ -0,0 +1,165 @@
+/**
+ * @license
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import difflib from 'difflib';
+
+import {css, customElement, html, LitElement, property} from 'lit-element';
+import {classMap} from 'lit-html/directives/class-map';
+import {styleMap} from 'lit-html/directives/style-map';
+
+import {styles} from './text_diff.css';
+
+interface TextDiff {
+  inputStrings: string[];
+  outputStrings: string[];
+  equal: boolean[];
+}
+
+
+/**
+ * A component that renders the diff between two strings, with the differences
+ * highlighted.  It can show before before and after, or only after.
+ */
+@customElement('lit-text-diff')
+export class TextDiffComponent extends LitElement {
+  /* The two text values to diff */
+  @property({type: String}) beforeText = undefined;
+  @property({type: String}) afterText = undefined;
+
+  /* Show a label next to the before/after text itself */
+  @property({type: String}) beforeLabel = undefined;
+  @property({type: String}) afterLabel = undefined;
+
+  /* Allow only rendering before or after, instead of both */
+  @property({type: Boolean}) includeBefore = true;
+  @property({type: Boolean}) includeAfter = true;
+
+  /* Optional: Diff by character instead of by word */
+  @property({type: Boolean}) byCharacter = false;
```

Comment:
The only changes here are these additional config options, to allow re-using this in the perturbations table.